### PR TITLE
Website published -> Website master

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

File: xtext-website/documentation/310_eclipse_support.md [hunk 1]
```diff
@@ -236,6 +236,13 @@ public void createReferenceType(Issue issue, IssueResolutionAcceptor acceptor) {
 
 Hence, there is the [ISyntaxErrorMessageProvider]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/parser/antlr/ISyntaxErrorMessageProvider.java) to assign issue codes to syntactical errors.
 
+## Auto Editing {#autoediting}
+Xtext-based editors automatically assist the user by inserting/deleting certain text on typing. E.g. inserting/removing closing single quotes, double quotes, parenthesis, square brackets or curly braces when the user inserts/removes the opening ones. Moreover, the auto-indentation functionality ensures the indentation-awareness of new lines: after hitting the `ENTER` key e.g. in a block enclosed by curly braces the cursor is automatically placed on the indented position in the subsequent new line.
```
Comment (tivervac) [inline xtext-website/documentation/310_eclipse_support.md:240]: s/on typing/at typetime/

s/. E.g/e.g/

File: xtext-website/documentation/310_eclipse_support.md [hunk 2]
```diff
@@ -482,6 +489,61 @@ public Class<? extends IComparator>
 
 Xtext also provides a quick outline: If you press CTRL-O in an Xtext editor, the outline of the model is shown in a popup window. The quick outline also supports drill-down search with wildcards. To enable the quick outline, you have to put the [QuickOutlineFragment2]({{site.src.xtext_core}}/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/outline/QuickOutlineFragment2.xtend) into your workflow.
 
+## Folding {#folding}
+
+Xtext calculates the editor folding regions based on the grammar out-of-the-box. Althought it comes with good defaults, sometimes they do not satisfy the needs and have to be customized.
```
Comment (tivervac) [inline xtext-website/documentation/310_eclipse_support.md:494]: s/Althought/Although/

File: xtext-website/documentation/310_eclipse_support.md [hunk 3]
```diff
@@ -482,6 +489,61 @@ public Class<? extends IComparator>
...
+
+Considering e.g the Xtext Statemachine example, the framework provides folding capabilities for the `state` regions:
+
+![](images/folding_default.png)
+
+In order to make `events`, `resetEvents` and `commands` foldable, too, a custom implementation of the [DefaultFoldingRegionProvider]({{site.src.xtext_eclipse}}/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/folding/DefaultFoldingRegionProvider.java) is necessary:
```
Comment (tivervac) [inline xtext-website/documentation/310_eclipse_support.md:500]: s/, too,/ too/

**Comment by cdietrich** [comment]
@miklossy can you please fix the typos in the original codebase?

**Comment by miklossy** [comment]
Yes.

**Pull request merged**